File: mobilepayments/src/main/java/com/begateway/mobilepayments/TransactionType.java
package com.begateway.mobilepayments;


public enum TransactionType {

    PAYMENT("PAYMENT", 0),
    AUTHORIZATION("AUTHORIZATION", 1),
    VERIFY("VERIFY", 2);

    private String stringValue;
    private int intValue;
    private TransactionType(String toString, int value) {
        stringValue = toString;
        intValue = value;
    }

    @Override
    public String toString() {
        return stringValue;
    }

}

File: mobilepayments/src/main/java/com/begateway/mobilepayments/models/network/request/TransactionType.kt
package com.begateway.mobilepayments.models.network.request

import com.google.gson.annotations.SerializedName

enum class TransactionType{
    @SerializedName("payment") PAYMENT,
    @SerializedName("authorization") AUTHORIZATION,
    @SerializedName("tokenization") TOKENIZATION
}
